client/boot/common.js

```diff
@@ -126,18 +129,24 @@ const loggedOutMiddleware = currentUser => {
 	}
 };
 
-const loggedInMiddleware = currentUser => {
+const loggedInMiddleware = ( currentUser, reduxStore ) => {
 	if ( ! currentUser.get() ) {
 		return;
 	}
 
 	page( '/', context => {
 		const { primarySiteSlug } = currentUser.get();
+		const state = reduxStore.getState();
+		const application = getApplication( state );
 		let redirectPath =
 			primarySiteSlug && 'variant' === abtest( 'redirectToCustomerHome' )
 				? `/home/${ primarySiteSlug }`
 				: '/read';
 
+		if ( 'jetpack-cloud' === application ) {
```

Comment:
It feels odd that this redirect is being handled in the logged-in middleware. It feels like it should be handled by its own middleware.

Reply:
yeah I am not a bit fan of this. The reader part and home redirects don't also make sense for me here. I wonder if this should be its own entry point.

Reply:
Will try to address this issue as a seperate PR from this. 
I think the whole loggedInMiddleware should be its own calypso section point that then redirects the user.